src/electionguard_cli/commands/e2e_command.py

```diff
@@ -45,43 +50,57 @@
     prompt_required=False,
 )
 @click.option(
-    "--output-file",
+    "--output-record",
     help="A file name for saving an output election record (e.g. './election.zip')."
     + " If no value provided then an election record will not be generated.",
     type=click.Path(exists=False),
     default=None,
 )
+@click.option(
+    "--output-keys",
+    help="A file name for saving the private and public guardian keys (e.g. './guardian-keys.json')."
+    + " If no value provided then no keys will be output.",
+    type=click.Path(exists=False, resolve_path=True),
+    default=None,
+)
 def e2e(
     guardian_count: int,
     quorum: int,
     manifest: TextIOWrapper,
     ballots: TextIOWrapper,
     spoil_id: str,
-    output_file: str,
+    output_record: str,
+    output_keys: str,
 ) -> None:
     """Runs through an end-to-end election."""
 
     # get user inputs
-    election_inputs = InputRetrievalStep().get_inputs(
-        guardian_count, quorum, manifest, ballots, spoil_id, output_file
+    election_inputs = E2eInputRetrievalStep().get_inputs(
+        guardian_count, quorum, manifest, ballots, spoil_id, output_record, output_keys
     )
 
     # perform election
-    joint_key = KeyCeremonyStep().run_key_ceremony(election_inputs)
-    build_election_results = ElectionBuilderStep().build_election(
+    joint_key = KeyCeremonyStep().run_key_ceremony(election_inputs.guardians)
+    build_election_results = ElectionBuilderStep().build_election_with_key(
         election_inputs, joint_key
     )
     submit_results = SubmitVotesStep().submit_votes(
         election_inputs, build_election_results
     )
-    decrypt_results = DecryptStep().decrypt_tally(
-        submit_results.data_store, election_inputs.guardians, build_election_results
+    (ciphertext_tally, spoiled_ballots) = TallyStep().get_from_ballot_store(
+        build_election_results, submit_results.data_store
```

Comment:
broke out tally work into it's own step since there was a lot of commonality between the two commands